### DISCUSSION TITLE
Stop a workflow

### DISCUSSION THREAD

**Discussion by mesfandi** [elsa-workflows/elsa-core#1480] [Q&A]
Hi,
When I start a work flow, is there any easy way to stop the workflow from the dashboard. I had a large workflow and I tried to delete the workflow instance, but it seems I cannot delete the instance while it's running. 
I have another question about the log files. When we have a large workflow with a "for loop" for example, it creates a lot of logs and it makes the dashboard really slow. when I load the workflow instances page, it takes a log time to load them and it cannot open a specific instance when I select it because the data of that workflow is large. Is there any way that we can limit the number of logs?
Thank you,
Javad

**Comment by sfmskywalker** [comment]
> When I start a work flow, is there any easy way to stop the workflow from the dashboard.

There is no way to do this from the dashboard currently, but it would be a good feature to have so I [created an issue](https://github.com/elsa-workflows/elsa-core/issues/1495) for it.

> I had a large workflow and I tried to delete the workflow instance, but it seems I cannot delete the instance while it's running.
This happens if the workflow instance was already in memory and is then later persisted again. If you keep trying to delete the workflow instance long enough, eventually you will win.

> I have another question about the log files. When we have a large workflow with a "for loop" for example, it creates a lot of logs and it makes the dashboard really slow. when I load the workflow instances page, it takes a log time to load them and it cannot open a specific instance when I select it because the data of that workflow is large. Is there any way that we can limit the number of logs?

Indeed. Right now, the workflow instance viewer loads ALL log entries. This should be updated such that the journal viewer loads one page at a time as you scroll down. I [created an issue](1496) for this as well.

Thanks for the questions!
